sympy/stats/frv_types.py

```diff
@@ -16,34 +16,36 @@
 from __future__ import print_function, division
 
 from sympy import (S, sympify, Rational, binomial, cacheit, Integer,
-        Dict, Basic, KroneckerDelta, Dummy)
+        Dict, Basic, KroneckerDelta, Dummy, And, Eq)
 from sympy.concrete.summations import Sum
 from sympy.core.compatibility import as_int, range
-from sympy.core.logic import fuzzy_not, fuzzy_and
+from sympy.stats.rv import _value_check
 from sympy.stats.frv import (SingleFinitePSpace, SingleFiniteDistribution)
 
 __all__ = ['FiniteRV', 'DiscreteUniform', 'Die', 'Bernoulli', 'Coin',
         'Binomial', 'Hypergeometric']
 
 def rv(name, cls, *args):
-    density = cls(*args)
-    return SingleFinitePSpace(name, density).value
+    args = list(map(sympify, args))
+    i = 0
+    while i < len(args): # Converting to Dict since dict is not hashable
+        if isinstance(args[i], dict):
+            args[i] = Dict(args[i])
+        i += 1
+    dist = cls(*args)
+    dist.check(*args)
+    return SingleFinitePSpace(name, dist).value
 
 class FiniteDistributionHandmade(SingleFiniteDistribution):
     @property
     def dict(self):
         return self.args[0]
 
-    def __new__(cls, density):
-        density = Dict(density)
-        for k in density.values():
-            k_sym = sympify(k)
-            if fuzzy_not(fuzzy_and((k_sym.is_nonnegative, (k_sym - 1).is_nonpositive))):
```

Comment:
looks like this was handling symbolic input. Does your modification allow the same?

Reply:
Yes. There tests for symbolic parameters which are passing, so I think my modifications doesn't cause any problem to those tests.

Reply:
@smichr Can I resolve this conversation?